src/service/request-service.js

```diff
@@ -2,14 +2,79 @@ const preconditions = require("preconditions").singleton();
 
 const AirtableUtils = require("../airtable-utils");
 const { logger } = require("../logger");
+const { getCoords } = require("../geo");
+const config = require("../config");
+const RequestRecord = require("../model/request-record");
 
 /**
  * APIs that deal with Request
  */
 class RequestService {
-  constructor(base) {
+  constructor(base, airtableUtils) {
     preconditions.shouldBeObject(base);
     this.base = base;
+    this.airtableUtils = airtableUtils;
+  }
+
+  /**
+   * Resolve and update coordinates for requester's address
+   * @param request {RequestRecord} Request requiring coordinates
+   * @returns {Promise<RequestRecord>} request records updated with coordinates
+   * @throws error when unable to resolve coordinates or update them in airtable
+   */
+  async resolveAndUpdateCoords(request) {
+    preconditions.shouldBeObject(request);
+    preconditions.shouldBeString(request.fullAddress);
+    try {
+      if (request.coordinates) {
```

Comment:
A request address could be updated, so there has to be similar logic to the other coordinate retrieval (which saves the address for which coordinates were calculated from).

Reply:
Just pushed a change addressing this change request.